src/python/pants/build_graph/target.py

```diff
@@ -687,31 +648,26 @@ def dependencies(self):
     return [self._build_graph.get_target(dep_address)
             for dep_address in self._build_graph.dependencies_of(self.address)]
 
-  def exports(self, dep_context):
-    """
-    :param dep_context: A DependencyContext with configuration for the request.
-
-    :return: targets that this target directly exports. Note that this list is not transitive,
-      but that exports are transitively expanded during the computation of strict_dependencies.
-    :rtype: list of Target
-    """
-    exports = self._cached_exports_map.get(dep_context, None)
+  @property
+  def export_addresses(self):
+    exports = self._cached_exports_addresses
     if exports is None:
-      exports = []
-      for export in getattr(self, 'export_specs', []):
-        if not isinstance(export, Target):
-          export_spec = export
-          export_addr = Address.parse(export_spec, relative_to=self.address.spec_path)
-          export = self._build_graph.get_target(export_addr)
-          if export not in self.dependencies:
-            # A target can only export its dependencies.
-            raise TargetDefinitionException(
-                self,
-                'Invalid export: "{}" must also be a dependency.'.format(export_spec))
-        if isinstance(export, dep_context.codegen_types):
-          export = _get_synthetic_target(self, export)
-        exports.append(export)
-      self._cached_exports_map[dep_context] = exports
+
+      exports = tuple()
```

Comment:
I don't expect tuple appends to be very fast, so this looks n^2 maybe... probably create a mutable list and then convert to a tuple at the end.